src/shapes.jl

```diff
@@ -32,7 +32,29 @@ function _uppertriang_to_vec(m::AbstractMatrix)
     reduce(vcat, [m[1:i, i] for i in 1:size(m, 1)])
 end
 
+function ChainRulesCore.rrule(::typeof(MGVI._uppertriang_to_vec), m::AbstractMatrix)
```

Comment:
I use uppertriang_to_vec because only half of the covariance parameters is independent.